core/frontend/src/main/scala/io/udash/properties/Property.scala

```diff
@@ -117,6 +140,74 @@ trait ReadableProperty[A] {
   }
 }
 
+abstract class BaseReadableSeqPropertyFromSingleValue[A, B : ModelValue](origin: ReadableProperty[A], transformer: A => Seq[B],
+                                                                         override val executionContext: ExecutionContext)
+  extends ReadableSeqProperty[B, ReadableProperty[B]] {
+
+  override val id: UUID = UUID.randomUUID()
+  override protected[properties] val parent: Property[_] = null
+  protected val structureListeners: mutable.Set[Patch[Property[B]] => Any] = mutable.Set()
+
+  val pc = implicitly[PropertyCreator[B]]
+  protected val children = mutable.ListBuffer.empty[Property[B]]
+
+  update(origin.get)
+  origin.listen(update)
+
+  private def structureChanged(patch: Patch[Property[B]]): Unit =
+    structureListeners.foreach(_.apply(patch))
+
+  private def update(v: A): Unit = {
+    val transformed = transformer(v)
+    val current = get
+    val commonBegin = {
```

Comment:
I'm thinking about a more "Scala" way of doing this, I'm not sure it's more readable and didn't test too much. Check my draft, build on it or ignore it:
```scala
def commonIdx(s1: Seq[B], s2: Seq[B]): Int = {
  val (longer, shorter) = if(s1.size > s2.size) (s1, s2) else (s2, s1)
  math.max(0, longer.zipWithIndex.indexWhere { case ((elem, idx)) => idx >= shorter.size || elem != shorter(idx) })
}
val commonBegin = commonIdx(transformed, current)
val commonEnd = commonIdx(transformed.reverse, current.reverse)
```